{/* Start Typescript Errors */}
All SDK methods return a response object or throw an error. If Error objects are specified in your OpenAPI Spec, the SDK will throw the appropriate Error type.



Validation errors can also occur when either method arguments or data returned from the server do not match the expected format. The `SDKValidationError` that is thrown as a result will capture the raw value that failed validation in an attribute called `rawValue`. Additionally, a `pretty()` method is available on this error that can be used to log a nicely formatted string since validation errors can list many issues and the plain error string may be difficult read when debugging. 


```typescript
import { HathoraCloud } from "@hathora/cloud-sdk-typescript";
import * as errors from "@hathora/cloud-sdk-typescript/models/errors";

async function run() {
    const sdk = new HathoraCloud({
        hathoraDevToken: "<YOUR_BEARER_TOKEN_HERE>",
        appId: "app-af469a92-5b45-4565-b3c4-b79878de67d2",
    });

    let result;
    try {
        result = await sdk.appV1.createApp({
            authConfiguration: {
                google: {
                    clientId: "<value>",
                },
                nickname: {},
                anonymous: {},
            },
            appName: "minecraft",
        });
    } catch (err) {
        switch (true) {
            case err instanceof errors.SDKValidationError: {
                // Validation errors can be pretty-printed
                console.error(err.pretty());
                // Raw value may also be inspected
                console.error(err.rawValue);
                return;
            }
            case err instanceof errors.ApiError: {
                console.error(err); // handle exception
                return;
            }
            default: {
                throw err;
            }
        }
    }

    // Handle the result
    console.log(result);
}

run();

```
{/* End Typescript Errors */}
